src/zeroband/training/utils.py

```diff
@@ -155,62 +53,33 @@ def __len__(self):
         return self.vocab_size
 
 
-class MetricsAverager:
-    """
-    Simple class that keep track of metrics over multiple gradient accumulation steps and sync across gpu when calling sync()
-    """
-
-    def __init__(self):
-        self.metrics = {}
-        self.count = {}
-        self.world_info = get_world_info()
-
-    @torch.no_grad()
-    def update(self, key, value: torch.Tensor | list[torch.Tensor]):
-        if isinstance(value, torch.Tensor):
-            self._update(key, value)
-        else:
-            for v in value:
-                self._update(key, v)
-
-    def _update(self, key, value: torch.Tensor):
-        if key not in self.metrics:
-            self.metrics[key] = value
-            self.count[key] = 1
-        else:
-            self.metrics[key] += value
-            self.count[key] += 1
-
-    @torch.no_grad()
-    def sync(self):
-        for key in self.metrics:
-            value = self.metrics[key].clone()
-            count = torch.tensor(self.count[key])
-
-            dist.all_reduce(value.to("cuda"), op=dist.ReduceOp.SUM)
-            dist.all_reduce(count.to("cuda"), op=dist.ReduceOp.SUM)
-
-            value = value / count
-
-            self.metrics[key] = value
-
-    def __getitem__(self, key):
-        return self.metrics[key]
-
-    def items(self):
-        return self.metrics.items()
-
-
 def get_real_tensor(tensor: torch.Tensor | DTensor):
     if isinstance(tensor, DTensor):
         return tensor.to_local()
     return tensor
 
 
+def compute_logprobs(
+    model: Model,
+    input_ids: torch.Tensor,
+    position_ids: torch.Tensor,
+    temperature: float,
+) -> torch.Tensor:
+    logits: Float[torch.Tensor, "batch seq vocab"] = model(
+        input_ids=input_ids, position_ids=position_ids
+    ).logits.contiguous()
+
+    input_ids_shifted = input_ids[:, 1:]
+    logits_shifted = logits[:, :-1, :] / temperature
+    logprobs = selective_log_softmax(logits_shifted, input_ids_shifted)
+    del logits, logits_shifted
+    return logprobs
+
+
 OffloadedTensor: TypeAlias = list[tuple[torch.Tensor, int]]
 
 
-def offload_model_to_cpu(model: ModelType) -> OffloadedTensor:
+def offload_model_to_cpu(model: Model) -> OffloadedTensor:
```

Comment:
```suggestion
def offload_model_to_cpu(model: Model) -> OffloadedTensor:
```

this should be modelType as its a type